### PR TITLE
[do not merge ] Samba4

### PR DESCRIPTION
**Pull request by arete**

### CODE REVIEW

File: srcpkgs/heimdal/template [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ --without-capng
+ --without-sqlite3
+ --without-libintl
+ --without-openssl
+ --without-berkeley-db
+ --without-readline
+ --without-libedit
+ --without-hesiod
+ --without-x
+ --disable-heimdal-documentation \
```
Comment (maxice8) [inline srcpkgs/heimdal/template]: \ not requires

File: srcpkgs/heimdal/template [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+revision=1
+build_style=gnu-configure
+makedepends="ncurses-devel"
+short_desc="Kerberos 5, PKIX, GSS-API, SPNEGO, NTLM, MD5, SASL implementation"
+maintainer="Gerardo Di Iorio <arete74@gmail.com>"
+license="GPL-3"
+homepage="http://www.h5l.org/"
+distfiles="https://github.com/heimdal/heimdal/releases/download/heimdal-$version/heimdal-$version.tar.gz"
+checksum=c5a2a0030fcc728022fa2332bad85569084d1c3b9a59587b7ebe141b0532acad
+configure_args="
```
Comment (maxice8) [inline srcpkgs/heimdal/template]: put after build_style=

File: srcpkgs/heimdal/template [hunk 3]
```diff
@@ -0,0 +1,45 @@
+# Template file for 'heimdal'
+pkgname=heimdal
+version=7.5.0
+revision=1
+build_style=gnu-configure
+makedepends="ncurses-devel"
+short_desc="Kerberos 5, PKIX, GSS-API, SPNEGO, NTLM, MD5, SASL implementation"
+maintainer="Gerardo Di Iorio <arete74@gmail.com>"
+license="GPL-3"
```
Comment (maxice8) [inline srcpkgs/heimdal/template]: please use SPDX short-identifier

File: srcpkgs/heimdal/template (hunk 2 above)
Comment (maxice8) [reply srcpkgs/heimdal/template]: and put it normally, side by side, cutting at 80 columns

File: srcpkgs/samba4/template [hunk 4]
```diff
@@ -0,0 +1,98 @@
+# Template file for 'samba'
+pkgname=samba4
+version=4.9.4
+revision=1
+build_style=waf
+configure_script=buildtools/bin/waf
+configure_args="--sysconfdir=/etc/samba \
```
Comment (maxice8) [inline srcpkgs/samba4/template]: no need for backslash, it is inside a quoted variable

Comment (maxice8) [reply srcpkgs/samba4/template]: also apply same pattern as used everywhere else for configure_args

File: srcpkgs/samba4/template [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+ --with-ads \
+ --with-shared-modules=idmap_ad,idmap_rid,idmap_adex,idmap_hash,idmap_tdb2,pdb_tdbsam,pdb_ldap,pdb_ads,pdb_smbpasswd,pdb_wbc_sam,pdb_samba4,auth_unix,auth_wbc,auth_server,auth_netlogond,auth_script,auth_samba4
+ --enable-cups \
+ --without-gettext 
+"
+patch_args="-Np1"
+wrksrc="samba-$version"
+short_desc="SMB/CIFS file, print, and login server for Unix"
+maintainer="Gerardo Di Iorio <arete74@gmail.com>"
+license="GPL-3"
```
Comment (maxice8) [inline srcpkgs/samba4/template]: SPDX short-identifier

File: srcpkgs/samba4/template [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+homepage="http://www.samba.org"
+distfiles="https://download.samba.org/pub/samba/stable/samba-$version.tar.gz"
+checksum=6d98a8d8bcccbe788e4bbb406362e6676311aca711a3f3cc9b3a404bb9ff0b4f
+CFLAGS="-DUSE_SETUIDX=0"
+conflicts="samba>=0"
+provides="samba-${version}_${revision}"
+conf_files="/etc/pam.d/samba /etc/samba/smb.conf"
+make_dirs="/etc/samba/private 0750 root root"
+hostmakedepends="pkg-config perl python docbook2x"
```
Comment (maxice8) [inline srcpkgs/samba4/template]: move them directly below configure_args and have pkgconf instead of pkg-config

File: srcpkgs/samba4/template [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+makedepends="readline-devel libcap-devel popt-devel e2fsprogs-devel
+ gnutls-devel python-devel
+ libldap-devel pam-devel acl-devel avahi-libs-devel tdb-devel talloc-devel cups-devel lmdb-devel jansson-devel gpgme-devel libarchive-devel"
+
+if [ "$CROSS_BUILD" ]; then
+	configure_args+=" --bundled-libraries=!asn1_compile,!compile_et"
+	hostmakedepends+="heimdal-devel"
```
Comment (maxice8) [inline srcpkgs/samba4/template]: when appendng leave a space before

File: srcpkgs/heimdal/template [hunk 8]
```diff
@@ -0,0 +1,32 @@
+# Template file for 'heimdal'
+pkgname=heimdal
```
Comment (maxice8) [inline srcpkgs/heimdal/template]: This package has content headers of various other packages and binaries as well, it is not suitable for inclusion

File: srcpkgs/samba4/template [hunk 9]
```diff
@@ -0,0 +1,85 @@
+# Template file for 'samba4'
+pkgname=samba4
+version=4.9.4
+revision=1
+wrksrc="samba-$version"
+build_style=waf
+configure_args="--sysconfdir=/etc/samba --with-configdir=/etc/samba --localstatedir=/var --enable-fhs --with-lockdir=/var/cache/samba --with-piddir=/var/run/samba --with-logfilebase=/var/log/samba --without-pam --without-systemd --with-ads --with-shared-modules=idmap_ad,idmap_rid,idmap_adex,idmap_hash,idmap_tdb2,pdb_tdbsam,pdb_ldap,pdb_ads,pdb_smbpasswd,pdb_wbc_sam,pdb_samba4,auth_unix,auth_wbc,auth_server,auth_netlogond,auth_script,auth_samba4 --enable-cups --without-gettext"
```
Comment (maxice8) [inline srcpkgs/samba4/template]: cut at 80 cols

File: srcpkgs/samba4/template [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+configure_script=buildtools/bin/waf
+conf_files="/etc/pam.d/samba /etc/samba/smb.conf"
+make_dirs="/etc/samba/private 0750 root root"
+hostmakedepends="pkgconf perl python docbook2x"
```
Comment (maxice8) [inline srcpkgs/samba4/template]: pkg-config

File: srcpkgs/samba4/template [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+makedepends="readline-devel libcap-devel popt-devel e2fsprogs-devel
+ gnutls-devel python-devel
+ libldap-devel pam-devel acl-devel avahi-libs-devel tdb-devel talloc-devel cups-devel lmdb-devel jansson-devel gpgme-devel libarchive-devel"
```
Comment (maxice8) [inline srcpkgs/samba4/template]: can be reduced in size by adding to the lines above

File: srcpkgs/samba4/template [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+distfiles="https://download.samba.org/pub/samba/stable/samba-$version.tar.gz"
+checksum=6d98a8d8bcccbe788e4bbb406362e6676311aca711a3f3cc9b3a404bb9ff0b4f
+provides="samba-${version}_${revision}"
+conflicts="samba>=0"
+CFLAGS="-DUSE_SETUIDX=0"
+patch_args="-Np1"
+
+if [ "$CROSS_BUILD" ]; then
+	configure_args+=" --bundled-libraries=!asn1_compile,!compile_et"
+	hostmakedepends+=" heimdal-devel"
```
Comment (maxice8) [inline srcpkgs/samba4/template]: please do not include this package

File: srcpkgs/samba4/template [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+		vmove "usr/lib/samba/*.so*"
+	}
+}
+samba4-devel_package() {
+	depends="libsmbclient>=${version}_${revision}"
+	short_desc+=" - development files"
+	pkg_install() {
+		vmove usr/include
+		vmove usr/lib/pkgconfig
+		vmove usr/lib/*.so
```
Comment (maxice8) [inline srcpkgs/samba4/template]: quote

File: srcpkgs/samba4/template [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+				vmove usr/bin/${g}
+		fi
+		done
+	}
+}
+libsamba4_package() {
+	short_desc="Shared libraries for communication with SMB/CIFS servers"
+	pkg_install() {
+		vmove "usr/lib/*.so.*"
+		vmove "usr/lib/samba/*.so*"
```
Comment (maxice8) [inline srcpkgs/samba4/template]: can be usr/lib/samba

**Comment by Anachron** [comment]
@arete please fix remaining issues.

File: srcpkgs/samba4/template (hunk 9 above)
Comment (Anachron) [reply srcpkgs/samba4/template]: Why compile it without pam (`--without-pam`) but then quire it later?

File: srcpkgs/samba4/template (hunk 12 above)
Comment (arete) [reply srcpkgs/samba4/template]: heimdal-devel is use only for cross compile. I try to disable heimdal build wthout success.

Comment (maxice8) [reply srcpkgs/samba4/template]: regardless, it conflicts with lots of packages.

Comment (maxice8) [reply srcpkgs/samba4/template]: use system-mit-krb5 with experimental ad-dc

**Comment by maxice8** [comment]
ping

**Comment by Anachron** [comment]
@arete please fix remaining issues.

**Comment by Anachron** [comment]
Tried to build this locally but I get a ton of errors for unresolved dependencies and alike.

**Comment by arete** [comment]
Hi, i try to  work to update at last version 4.11.

**Comment by Anachron** [comment]
Please update this PR, I will try to help as good as possible because I urgently need it.

**Comment by Anachron** [comment]
Thanks for the new commits, I'm building it right now and will test it on my raspberry pi 3 B+.

**Comment by Gottox** [comment]
I need to get back to this soon. Keeping it open for now.

**Comment by Anachron** [comment]
@Gottox @arete ping.

**Comment by Anachron** [comment]
Since `libnetapi.0.so` is both provided by the `samba` and `samba4` package we came to the conclusion that it's probably best to just upgrade the samba package to the version 4.

I'm preparing such a PR right now.

**Comment by damadmai** [comment]
Hi @Anachron Do you already have a pull request for samba4?
Could I help somehow?

**Comment by Gottox** [comment]
outdated

**Pull request closed without merging**